apps/mobile/src/libs/components/wheel/libs/components/animated-appearing-sector/animated-appearing-sector.tsx

```diff
@@ -0,0 +1,96 @@
+import { Animated, Circle } from "~/libs/components/components";
+import { useSharedValue } from "~/libs/hooks/hooks";
+
+import { useWheelAnimation } from "../../hooks/hooks";
+
+type Properties = {
+	animationRepetitions: number;
+	animationTime: number;
+	centerGap: number;
+	centerPoint: number;
+	endPercentInner: number;
+	endPercentOuter: number;
+	height: number;
+	layerOffset: number;
+	outlineColor: string;
+	sectorColor: string;
+	startPercentInner: number;
+	startPercentOuter: number;
+};
+
+const INITIAL_ANIMATED_VALUE = 0;
+const AnimatedCircle = Animated.createAnimatedComponent(Circle);
+
+const AnimatedAppearingSector: React.FC<Properties> = ({
+	animationRepetitions,
+	animationTime,
+	centerGap,
+	centerPoint,
+	endPercentInner,
+	endPercentOuter,
+	height,
+	layerOffset,
+	outlineColor,
+	sectorColor,
+	startPercentInner,
+	startPercentOuter,
+}: Properties) => {
+	const animatedInnerArrayDash = useSharedValue(INITIAL_ANIMATED_VALUE);
+	const animatedInnerArrayGap = useSharedValue(INITIAL_ANIMATED_VALUE);
+	const animatedInnerDashOffset = useSharedValue(INITIAL_ANIMATED_VALUE);
+	const animatedInnerStrokeWidth = useSharedValue(INITIAL_ANIMATED_VALUE);
+
+	const animatedOuterArrayDash = useSharedValue(INITIAL_ANIMATED_VALUE);
+	const animatedOuterArrayGap = useSharedValue(INITIAL_ANIMATED_VALUE);
+	const animatedOuterDashOffset = useSharedValue(INITIAL_ANIMATED_VALUE);
+	const animatedOuterStrokeWidth = useSharedValue(INITIAL_ANIMATED_VALUE);
+
+	const animatedRadius = useSharedValue(INITIAL_ANIMATED_VALUE);
```

Comment:
I guess we can move the declaration of shared values to the hook, and pass only initialSharedValues object

Reply:
I pass the `initialSharedValues` as an argument, and inside the hook iterate through the keys, adding every key to the `sharedValues` declared in the hook? Or is there a better approach?

Reply:
You pass `sectorSharedValues` to the hook, but you need to declare all those values inside of the hook